core/src/test/java/org/elasticsearch/index/query/RegexpQueryBuilderTest.java

```diff
@@ -0,0 +1,104 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class RegexpQueryBuilderTest extends BaseQueryTestCase<RegexpQueryBuilder> {
+
+    private static final String[] REWRITE_METHOD = new String[]{"constant_score_auto", "scoring_boolean", "constant_score_boolean", 
+            "constant_score_filter", "top_terms_boost_1", "top_terms_1"};
+    
+    @Override
+    protected RegexpQueryBuilder doCreateTestQueryBuilder() {
+        RegexpQueryBuilder query;
+
+        // mapped or unmapped field
+        String text = randomAsciiOfLengthBetween(1, 10);
+        if (randomBoolean()) {
+            query = new RegexpQueryBuilder(STRING_FIELD_NAME, text);
+        } else {
+            query = new RegexpQueryBuilder(randomAsciiOfLengthBetween(1, 10), text);
+        }
+        if (randomBoolean()) {
+            List<RegexpFlag> flags = new ArrayList<>(); // used to test combination of flags
+            for (RegexpFlag flag : RegexpFlag.values()) {
+                if (randomBoolean()) {
+                    flags.add(flag);
+                }
+            }
+            query.flags(randomFrom(flags));
+        }
+        if (randomBoolean()) {
+            query.maxDeterminizedStates(randomInt(50000));
+        }
+        if (randomBoolean()) {
+            query.rewrite(randomFrom(REWRITE_METHOD));
+        }
+
+        return query;
+    }
+
+    @Override
+    protected Query doCreateExpectedQuery(RegexpQueryBuilder queryBuilder, QueryParseContext context) throws IOException {
+        MultiTermQuery.RewriteMethod method = QueryParsers.parseRewriteMethod(queryBuilder.rewrite(), null);
+
+        Query query = null;
+        MappedFieldType fieldType = context.fieldMapper(queryBuilder.fieldName());
+        if (fieldType != null) {
+            query = fieldType.regexpQuery(queryBuilder.value(), queryBuilder.flags(), queryBuilder.maxDeterminizedStates(), method, context);
+        }
+        if (query == null) {
+            RegexpQuery regexpQuery = new RegexpQuery(new Term(queryBuilder.fieldName(), BytesRefs.toBytesRef(queryBuilder.value())), 
+                    queryBuilder.flags(), queryBuilder.maxDeterminizedStates());
+            if (method != null) {
+                regexpQuery.setRewriteMethod(method);
+            }
+            query = regexpQuery;
+        }
+        return query;
+    }
+
+    @Test
+    public void testValidate() {
+        RegexpQueryBuilder commonTermsQueryBuilder = new RegexpQueryBuilder("", "regex");
+        assertThat(commonTermsQueryBuilder.validate().validationErrors().size(), is(1));
+
+        commonTermsQueryBuilder = new RegexpQueryBuilder("field", null);
+        assertThat(commonTermsQueryBuilder.validate().validationErrors().size(), is(1));
+
+        commonTermsQueryBuilder = new RegexpQueryBuilder("field", "regex");
+        assertNull(commonTermsQueryBuilder.validate());
```

Comment:
very minor comment, but with randomization we could also test the case when there are multiple errors. Just randomize the builder that you create, based on what you set to it keep track of how many errors you expect, then base the assertion on it. A single randomized builder creation and a single assertion at the end that will cover all the cases.